docker-images/openemr/run_openemr.sh

```diff
@@ -32,7 +32,33 @@ auto_setup() {
     chmod -R 600 .
     php auto_configure.php -f ${CONFIGURATION} || return 1
 
-    echo "OpenEMR configured. Setting user 'www' as owner of openemr/ and setting file/dir permissions to 400/500"
+    echo "OpenEMR configured."
+    CONFIG=$(php -r "require_once('/var/www/localhost/htdocs/openemr/sites/default/sqlconf.php'); echo \$config;")
+    if [ "$CONFIG" == "0" ]; then
+        echo "Error in auto-config. Configuration failed."
+        exit 2
+    fi
+}
+
+CONFIG=$(php -r "require_once('/var/www/localhost/htdocs/openemr/sites/default/sqlconf.php'); echo \$config;")
+if [ "$CONFIG" == "0" ] &&
+   [ "$MYSQL_HOST" != "" ] &&
+   [ "$MYSQL_ROOT_PASS" != "" ] &&
+   [ "$MANUAL_SETUP" != "yes" ]; then
+
+    echo "Running quick setup!"
+    while ! auto_setup; do
+        echo "Couldn't set up. Any of these reasons could be what's wrong:"
+        echo " - You didn't spin up a MySQL container or connect your OpenEMR container to a mysql instance"
+        echo " - MySQL is still starting up and wasn't ready for connection yet"
+        echo " - The Mysql credentials were incorrect"
+        sleep 1;
+    done
+    echo "Setup Complete!"
+fi
+
+if [ "$CONFIG" == "1" ]; then
```

Comment:
Doesn't all the stuff also need to happen after the autoinstall?
(I think above if will skip that scenario)

Reply:
I double checked. It does correctly happen after autoinstall (since autoinstall now sets CONFIG again when it completes)

Reply:
oh, that makes sense (forgot that a function doesn't limit scope of a variable in bash). looks good to go then.